### DISCUSSION TITLE
Welcome to the new discussion area!

### DISCUSSION THREAD

**Discussion by qreeves** [redeclipse/base#1136] [General]
So, you've probably ended up here expecting to find the [old forum](https://redeclipse.net/oldforum), but surprise! We're now using this place as a means to conduct our discussions and share content with each other. All posts here will also be relayed to our [IRC channel](https://redeclipse.net/chat) on `irc.freenode.net` in `#redeclipse`.

We're also working on porting our [old wiki](https://redeclipse.net/wiki) to GitHub pages, which requires converting the old MediaWiki pages to Markdown. You can get involved by porting pages from the [old wiki repository](https://github.com/red-eclipse/oldwiki) to [here](https://github.com/red-eclipse/docs). Hopefully, we can take advantage of the features and development behind Markdown to produce a proper user manual (and maybe even the future website).

If you're wondering what prompted this, well, it has been something I've been thinking about for a while, especially as I grew accustomed to GitHub and grew to appreciate their reliability. The old forums are a heavily modified and outdated version of phpBB and the wiki has the same inherent problem with MediaWiki.

It might be a little clunky to use an issue tracker for a forum, but it is more reliable and gives us greater flexibility, while removing the need for us to take time out from development in order to maintain essential project tools. I personally welcome our Markdown overlords and thank them for providing us with these awesome development tools.

You can find the old forum [here](https://redeclipse.net/oldforum).

**Comment by honno** [comment]
I'm really interested in this change! The commit history conversion with the old wiki is really neat, and the auto formatting for markdown seems real useful. Obviously it's a hit-and-miss so I'll see about doing fixing some up and adding them to the wiki here :)

**Comment by SniperGoth** [comment]
While i preffer having the forum itself, let's see how this change goes on down the line....

**Comment by cloudziq** [comment]
interesting, hard to tell what emerge from that change, hope there will be no cluttering in these issues (ah just noticed that here we can sort everything based on labels or such)

**Comment by IceflowRE** [comment]
The wiki here may be good, but the forum? lets see.
However transfering the wiki here, i will be there.

**Comment by ZeroKnight** [comment]
Looking forward to seeing how this turns out; very interesting idea. Plus, posts relayed to IRC for free is pretty dope 👍

**Comment by mrmistah** [comment]
Its gonna be weird for me to discuss using github since I rarely use it, but I think it'll be good for to start knowing the ropes.

Ill see what I can do with the wikipages.....I just gotta learn markdown first.

**Comment by poVoq** [comment]
If it doesn't work out you are of course also always free to come back to freegamedev.net (but we also really need to update the forum software at some point...)

**Comment by poVoq** [comment]
@acerspyro I agree in general, but a forum is not really a vital part of the game. A forum also benefits from having an existing user-base.

Discourse has the disadvantage that it is quite resource heavy and doesn't run on a typical web-host. An up to date phpBB is really not that bad. Otherwise, Flarum and NodeBB are also cool options.

**Comment by qreeves** [comment]
None of this addresses the fundamental problem, which is the fact that these things require a dedicated server, maintenance, and quite often heavy customisation. Customising things pretty much ensures you're going to have issues upgrading it, leaving you with security holes.

We don't have a web developer, and even if we did it would be so temporary that I'm lumped doing everything in that regard in the end anyway. In the ten years I've been doing this, I have not been happy with any of the solutions we've tried, and I'm honestly sick and tired of trying to fix "web component X".

When it comes to third party hosting, it comes down to a matter of trust, especially after the fall of Sourceforge. GitHub is the only place that has not only demonstrated a level of that trust, but also a consistent drive to continue developing the platform. Gamejolt may have forums for projects soon, but I'm not as confident in their long-term viability.

Typically, forum posts end up buried by time anyway, and anyone interested in following it further can subscribe to the topic (or the whole repository) to receive updates via notification or email. I don't think it is a stretch to just allow the creation of more new topics, even if they update old ones, instead of bumping the aforementioned old posts. Especially when it comes to support requests, they don't last long anyway.

There are sort options, which would allow sorting by last updated. Maybe we can make the redirection links include the proper GET parameters so everyone coming to the page would have it sorted by default. Perhaps we can also convince GitHub to give us a Discuss section based on Issues, or allow us to specify the Issues sort order in the repository settings. There's actually quite a few projects out there doing the same thing we're doing here with Issues.

**Comment by ghost** [comment]
In the past I've used ProBoards when I've needed forums (like team/clan forums). It's free (to use) and provides most of the things you want in forum software, without the headaches of administrating it (just moderating).

https://proboards.com/

I'm not a fan of using GitHub as a forum, as we've lost a lot functionality we once had (post approval, signatures, moderation panel, reporting posts, searching and categorisation.

**Comment by Calinou** [comment]
> In the past I've used ProBoards when I've needed forums (like team/clan forums). It's free (to use) and provides most of the things you want in forum software, without the headaches of administrating it (just moderating).

Unfortunately, it comes with advertisements, whereas GitHub has never displayed any ads, not to mention the domain name doesn't give it a very trustworthy feeling.

**Comment by ghost** [comment]
The [Login with GitHub](http://www.redeclipse.net/issues#) link on the discussion page is broken.
